docs/README.md

```diff
@@ -1,112 +1,132 @@
-It run as a separate routine. It is responsible for looking at the services and storing the information in the store.  
+Proxless works with 4 core concepts:
 
-### Get services
+- the **store** is an in-memory storage containing the information for each route.
+- the **proxy**, responsible for forwarding the requests to the correct back end and scaling up the deployment if needed.
+- the **services engine**, responsible for retrieving the configuration from the services and configuring the deployments.
+- the **downscaler**, responsible for downscaling the deployments when they are not used.
 
-It will look for any kubernetes services with the 2 annotations
+### The store
 
-- `proxless/domains` - external domain names associated to the service (separated with `,`) - example: `proxless/domains=example.io,www.example.io` 
-- `proxless/deployment` - deployment name used for scaling up and down the app - example: `hello-world`
+The store is an in-memory map and contain the information for each route.  
+It will contain
 
-### Label deployment
+- the namespace name where the deployment and service are
+- the service name proxless will forward the request to
+- the deployment name proxless will scale up and down
+- the domain names / urls proxless proxless will proxy
+- a timestamp of the last time the service has been requested
 
-It will label the associated deployed with `proxless=true`.  
-This will be used by the downscaler to get all the deployment managed by proxless.
+The logic of the store is available in [internal/store/inmemory/inmemory.go](../internal/store/inmemory/inmemory.go).
 
-### Add service information into the store
+### The Proxy
 
-It will create an object [`Route`](../internal/store/routes.go) and add it to an in-memory store (map) using the UUID of the service.  
+The Proxy is the main process of proxless. It is the one responsible for the liveness of proxless. If the proxy crash, proxless restart.  
+It is responsible for forwarding the requests to the correct back end and scaling up the deployment if needed.
 
-Example:
-```shell script
-uuid            -> Route Object containing the information of the service
+Upon receiving a request to a specific URL, the proxy will
 
-domain 1        -> uuid
-domain 2        -> uuid
+- retrieve the route information from the store
+    - if the route is not in the store, it will return a `404`
+- forward the request (with the headers) to the service
+    - if the call fail (`could not resolve host` error), it will immediate try to scale down the deployment
+    - when the deployment is ready, it will forward the request to the service
+    - it will also the `lastUsed` timestamp of the route in the store
+- if the above fail, it will return a `500`
 
-deployment name -> uuid
-```
+The logic of the proxy is available in [internal/server/http/http.go](../internal/server/http/http.go).
 
-## Proxy
+### The Services Engine
```

Comment:
comments - I think the docs needs to be split up in "how proxless works" and "how to deploy a service". There's a lot of information mixed here on the inner workings which is great, but its a lot to "dodge" in just getting a service up and running in k8s.  May want to continue to work on this, maybe read other readme's and see what style you like.  I think some mix of example, how it works, then deep dive (easy -> medium -> hard) is important.

Reply:
Yep Nadeem mentions that too, I'll add an easy "How to deploy"